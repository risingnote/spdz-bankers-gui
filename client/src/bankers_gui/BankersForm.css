.bankersForm {
  margin-right: 3rem;
  text-align: left;
}

.bankersForm label {
  display: block;
}

.bankersForm input {
  display: block;
  margin-bottom: 10px;
}

.diners {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2rem;
}

.atable {
  background: lightyellow;
  border: thin solid grey;
  width: 18rem;
  height: 18rem;
  border-radius: 50%;
  position: relative;
}

.aplace {
  background: white;
  border: thin solid grey;
  width: 4rem;
  height: 4rem;
  border-radius: 10%;
  position: absolute;
  /* centre in container */
  top: 50%;
  left: 50%;
  margin: -2rem 0 0 -2rem;
  /* centre contained text */
  text-align: center;
  vertical-align: middle;
  line-height: 4rem;
}

/* Translation is 
 * (radius.sinØ, -radiuscosØ)
*/