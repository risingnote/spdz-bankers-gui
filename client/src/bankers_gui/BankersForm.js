/**
 * Responsible for managing local state on form before submit.
 */
import React, { Component, PropTypes } from 'react'

import './BankersForm.css'

class BankersForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      bonus: '',
      participantName: ''
    }
    this.handleBonusChange = this.handleBonusChange.bind(this)
    this.handleNameChange = this.handleNameChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleBonusChange(event) {
    this.setState({bonus: event.target.value})
  }

  handleNameChange(event) {
    this.setState({participantName: event.target.value})
  }

  handleSubmit(event) {
    this.props.submitBonus(this.state.participantName, this.state.bonus)
    this.setState({bonus: '', participantName: ''})
    event.preventDefault()
  }

  render() {
    const disableSubmit = this.props.joinedName === undefined && this.props.connectionProblem === undefined ? '' : 'disabled'
    const statusMessage = () => {
      if (this.props.connectionProblem !== undefined) return this.props.connectionProblem
      else if (this.props.winnerChosen) return `A winner has been chosen.`
      else if (this.props.joinedName !== undefined) return `You have joined the meal as ${this.props.joinedName}.`
      else return ''
    }

    return (
        <form className="BankersForm" onSubmit={this.handleSubmit}>
          <h4>Bankers Celebration Dinner</h4>
          <p>...but who should pay?</p>        
          <label htmlFor="joinName">Join meal as</label>
          <input type="text" id="joinName" value={this.state.participantName} onChange={this.handleNameChange} disabled={disableSubmit}/>
          <label htmlFor="bonusValue">Bonus</label>
          <input type="text" id="bonusValue" value={this.state.bonus} onChange={this.handleBonusChange} disabled={disableSubmit}/>
          <input type="submit" value="Send" disabled={disableSubmit}/>
          <p className='smallText'>{statusMessage()}</p>
        </form>
    )
  }
}

BankersForm.propTypes = {
  submitBonus: PropTypes.func.isRequired,
  joinedName: PropTypes.string,
  winnerChosen: PropTypes.bool,
  connectionProblem: PropTypes.string
}

export default BankersForm
