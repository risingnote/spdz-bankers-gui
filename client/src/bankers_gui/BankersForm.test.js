import React from 'react'
import renderer from 'react-test-renderer'

import BankersForm from './BankersForm'

describe('Bankers Form display component rendering', () => {
  it('Renders as expected (compared to a snapshot) start of game.', () => {

    const tree = renderer.create(
      <BankersForm submitBonus={()=>{}} joinedName={undefined} winnerChosen={undefined} resetGame={()=>{}} connectionProblem={undefined} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Renders as expected (compared to a snapshot) start of game, no proxies.', () => {

    const tree = renderer.create(
      <BankersForm submitBonus={()=>{}} joinedName={undefined} winnerChosen={undefined} resetGame={()=>{}} connectionProblem={'No proxies found.'} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Renders as expected (compared to a snapshot) after user has joined game.', () => {

    const tree = renderer.create(
      <BankersForm submitBonus={()=>{}} joinedName={'Bob'} winnerChosen={undefined}  resetGame={()=>{}} connectionProblem={undefined}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Renders as expected (compared to a snapshot) after winner has been chosen.', () => {

    const tree = renderer.create(
      <BankersForm submitBonus={()=>{}} joinedName={'Bob'} winnerChosen={true}  resetGame={()=>{}} connectionProblem={undefined}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

})
