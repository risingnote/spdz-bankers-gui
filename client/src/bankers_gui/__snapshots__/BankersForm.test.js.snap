exports[`Bankers Form display component rendering Renders as expected (compared to a snapshot) after user has joined game. 1`] = `
<form
  className="BankersForm"
  onSubmit={[Function]}>
  <h4>
    Bankers Celebration Dinner
  </h4>
  <p>
    ...but who should pay?
  </p>
  <label
    htmlFor="joinName">
    Join meal as
  </label>
  <input
    disabled="disabled"
    id="joinName"
    onChange={[Function]}
    type="text"
    value="" />
  <label
    htmlFor="bonusValue">
    Bonus
  </label>
  <input
    disabled="disabled"
    id="bonusValue"
    onChange={[Function]}
    type="text"
    value="" />
  <input
    disabled="disabled"
    type="submit"
    value="Send" />
  <p
    className="smallText">
    You have joined the meal as Bob.
  </p>
</form>
`;

exports[`Bankers Form display component rendering Renders as expected (compared to a snapshot) after winner has been chosen. 1`] = `
<form
  className="BankersForm"
  onSubmit={[Function]}>
  <h4>
    Bankers Celebration Dinner
  </h4>
  <p>
    ...but who should pay?
  </p>
  <label
    htmlFor="joinName">
    Join meal as
  </label>
  <input
    disabled="disabled"
    id="joinName"
    onChange={[Function]}
    type="text"
    value="" />
  <label
    htmlFor="bonusValue">
    Bonus
  </label>
  <input
    disabled="disabled"
    id="bonusValue"
    onChange={[Function]}
    type="text"
    value="" />
  <input
    disabled="disabled"
    type="submit"
    value="Send" />
  <p
    className="smallText">
    A winner has been chosen.
  </p>
</form>
`;

exports[`Bankers Form display component rendering Renders as expected (compared to a snapshot) start of game, no proxies. 1`] = `
<form
  className="BankersForm"
  onSubmit={[Function]}>
  <h4>
    Bankers Celebration Dinner
  </h4>
  <p>
    ...but who should pay?
  </p>
  <label
    htmlFor="joinName">
    Join meal as
  </label>
  <input
    disabled="disabled"
    id="joinName"
    onChange={[Function]}
    type="text"
    value="" />
  <label
    htmlFor="bonusValue">
    Bonus
  </label>
  <input
    disabled="disabled"
    id="bonusValue"
    onChange={[Function]}
    type="text"
    value="" />
  <input
    disabled="disabled"
    type="submit"
    value="Send" />
  <p
    className="smallText">
    No proxies found.
  </p>
</form>
`;

exports[`Bankers Form display component rendering Renders as expected (compared to a snapshot) start of game. 1`] = `
<form
  className="BankersForm"
  onSubmit={[Function]}>
  <h4>
    Bankers Celebration Dinner
  </h4>
  <p>
    ...but who should pay?
  </p>
  <label
    htmlFor="joinName">
    Join meal as
  </label>
  <input
    disabled=""
    id="joinName"
    onChange={[Function]}
    type="text"
    value="" />
  <label
    htmlFor="bonusValue">
    Bonus
  </label>
  <input
    disabled=""
    id="bonusValue"
    onChange={[Function]}
    type="text"
    value="" />
  <input
    disabled=""
    type="submit"
    value="Send" />
  <p
    className="smallText">
    
  </p>
</form>
`;
